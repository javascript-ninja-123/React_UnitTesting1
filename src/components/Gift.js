import React,{Component} from 'react';
import { Input ,Form, Icon } from 'semantic-ui-react'

class GIFT extends Component {
    constructor(props){
    	super(props);
    	this.state = {person:'',present:''};
    }
    onChangePerson(e){
      this.setState({person:e.target.value})
    }
    onChangePresent(e){
      this.setState({present:e.target.value})
    }
    render() {
        return (
            <div className="GIFT">
              <Form>
                 <Form.Field>
                  <label>Person</label>
                  <Input
                  placeholder='person'
                  type='text'
                  value={this.state.person}
                  className='person-input'
                  onChange={this.onChangePerson.bind(this)}
                   />
                 </Form.Field>
                 <Form.Field>
                   <label>Present</label>
                    <Input
                    placeholder='present'
                    type='text'
                    className='present-input'
                    value={this.state.present}
                    onChange={this.onChangePresent.bind(this)}
                     />
                 </Form.Field>
                 <Icon
                 name='remove'
                 className='remove-btn'
                 onClick={() => this.props.removeClick(this.props.value.id)}
                 />
              </Form>
            </div>
        );
    }
}

export default GIFT
