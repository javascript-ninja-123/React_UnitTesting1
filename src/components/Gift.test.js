import React from 'react';
import {shallow} from 'enzyme';
import setupTest from '../setupTest';
import Gift from './Gift';

//Gift description
describe('Gift', () => {
  //define gift
  const mockRemove = jest.fn();
  const value = {id:1};
  const props = {value, removeClick:mockRemove}
  const gift = shallow(<Gift {...props}/>);

  //HTML description
  describe('HTML (static)', () => {
    it('renders it correctly', () => {
      expect(gift).toMatchSnapshot();
      expect(gift.exists()).toBe(true);
    })
    it('initiate a state with a person and a present', () => {
      expect(gift.state()).toEqual({person:'',present:''})
    })
  })
  //ends here

  //Javasript description
  describe('Javascript (behavior)', () => {
    //person input
    describe('when typing into person input', () => {
      const uncle = 'Uncle';
      beforeEach(() => {
        gift.find('.person-input').simulate('change',{target:{value:uncle}})
      })
      it('update the person in state', () => {
        expect(gift.state().person).toEqual(uncle)
      })
    })
  })
    //ends here

    //present input
    describe('when typing into present input', () => {
        const robot = 'robot';
      beforeEach(() => {
          gift.find('.present-input').simulate('change',{target:{value:robot}})
      })
      it('update the person in state', () => {
        expect(gift.state().present).toEqual(robot)
      })
    })
    //ends here

    //remove a gift Button
    describe('when clicking the Remove button', () => {
      beforeEach(() => {
        gift.find('.remove-btn').simulate('click')
      })
      it('calls the removeGift callback', () => {
        expect(mockRemove).toHaveBeenCalledWith(value.id)
      })
    })
    //ends here
})
//ends here
