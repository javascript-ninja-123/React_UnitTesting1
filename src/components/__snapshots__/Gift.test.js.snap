// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift HTML (static) renders it correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GIFT
    removeClick={[Function]}
    value={
        Object {
            "id": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <Form
        as="form"
>
        <FormField>
                <label>
                        Person
                </label>
                <Input
                        className="person-input"
                        onChange={[Function]}
                        placeholder="person"
                        type="text"
                        value=""
                />
        </FormField>
        <FormField>
                <label>
                        Present
                </label>
                <Input
                        className="present-input"
                        onChange={[Function]}
                        placeholder="present"
                        type="text"
                        value=""
                />
        </FormField>
        <Icon
                as="i"
                className="remove-btn"
                name="remove"
                onClick={[Function]}
        />
</Form>,
      "className": "GIFT",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "as": "form",
        "children": Array [
          <FormField>
            <label>
                        Person
            </label>
            <Input
                        className="person-input"
                        onChange={[Function]}
                        placeholder="person"
                        type="text"
                        value=""
            />
</FormField>,
          <FormField>
            <label>
                        Present
            </label>
            <Input
                        className="present-input"
                        onChange={[Function]}
                        placeholder="present"
                        type="text"
                        value=""
            />
</FormField>,
          <Icon
            as="i"
            className="remove-btn"
            name="remove"
            onClick={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <label>
                Person
</label>,
              <Input
                className="person-input"
                onChange={[Function]}
                placeholder="person"
                type="text"
                value=""
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Person",
              },
              "ref": null,
              "rendered": "Person",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "className": "person-input",
                "onChange": [Function],
                "placeholder": "person",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <label>
                Present
</label>,
              <Input
                className="present-input"
                onChange={[Function]}
                placeholder="present"
                type="text"
                value=""
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Present",
              },
              "ref": null,
              "rendered": "Present",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "className": "present-input",
                "onChange": [Function],
                "placeholder": "present",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "as": "i",
            "className": "remove-btn",
            "name": "remove",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <Form
          as="form"
>
          <FormField>
                    <label>
                              Person
                    </label>
                    <Input
                              className="person-input"
                              onChange={[Function]}
                              placeholder="person"
                              type="text"
                              value=""
                    />
          </FormField>
          <FormField>
                    <label>
                              Present
                    </label>
                    <Input
                              className="present-input"
                              onChange={[Function]}
                              placeholder="present"
                              type="text"
                              value=""
                    />
          </FormField>
          <Icon
                    as="i"
                    className="remove-btn"
                    name="remove"
                    onClick={[Function]}
          />
</Form>,
        "className": "GIFT",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "as": "form",
          "children": Array [
            <FormField>
              <label>
                            Person
              </label>
              <Input
                            className="person-input"
                            onChange={[Function]}
                            placeholder="person"
                            type="text"
                            value=""
              />
</FormField>,
            <FormField>
              <label>
                            Present
              </label>
              <Input
                            className="present-input"
                            onChange={[Function]}
                            placeholder="present"
                            type="text"
                            value=""
              />
</FormField>,
            <Icon
              as="i"
              className="remove-btn"
              name="remove"
              onClick={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <label>
                  Person
</label>,
                <Input
                  className="person-input"
                  onChange={[Function]}
                  placeholder="person"
                  type="text"
                  value=""
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Person",
                },
                "ref": null,
                "rendered": "Person",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "className": "person-input",
                  "onChange": [Function],
                  "placeholder": "person",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <label>
                  Present
</label>,
                <Input
                  className="present-input"
                  onChange={[Function]}
                  placeholder="present"
                  type="text"
                  value=""
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Present",
                },
                "ref": null,
                "rendered": "Present",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "className": "present-input",
                  "onChange": [Function],
                  "placeholder": "present",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "as": "i",
              "className": "remove-btn",
              "name": "remove",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
